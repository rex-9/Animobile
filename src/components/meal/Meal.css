.card {
  text-align: right;
  font-size: 18px;
}

.card img {
  opacity: 0.7;
  height: 200px;
  object-fit: cover;
}

.card > div {
  text-transform: uppercase;
  padding: 5px 10px;
}
