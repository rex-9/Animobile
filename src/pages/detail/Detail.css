.detail header {
  padding: 5px 10px;
  font-size: 18px;
  background-color: var(--medium);
}

.detail div {
  padding: 10px 20px;
}

.detail div:nth-child(2n+1) {
  background-color: #ce4176;
}
