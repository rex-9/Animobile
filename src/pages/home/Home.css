header {
  padding: 5px 10px;
  font-size: 18px;
  background-color: var(--medium);
}

.meals {
  display: grid;
  grid-template-columns: 1fr 1fr;
}
