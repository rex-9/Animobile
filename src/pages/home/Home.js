import Highlight from '../../components/highlight/Highlight';
import Nav from '../../components/nav/Nav';
import Meals from '../../components/meals/Meals';

const category = {
  img: 'https://www.themealdb.com/images/category/breakfast.png',
  name: 'Breakfast',
  flavour: 1000,
};

const Home = () => (
  <>
    <div className="home">
      <Nav navContent="Seafood" back={false} />
      <Highlight img={category.img} name={category.name} flavour={category.flavour} />
      <Meals />
    </div>
  </>
);

export default Home;
