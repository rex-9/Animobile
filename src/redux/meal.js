/* eslint-disable no-param-reassign */

import { createAsyncThunk } from '@reduxjs/toolkit';
import API from '../api';

const FILTER_MEALS = 'yummyverse/meal/FILTER_MEALS';
const DETAIL_MEAL = 'yummyverse/meal/DETAIL_MEAL';

const initialState = [];

// Meal Reducer
export default (state = initialState, action) => {
  let newState = {};
  let foundIndex = 0;
  switch (action.type) {
    case `${FILTER_MEALS}/fulfilled`:
      return [...action.payload];

    case DETAIL_MEAL:
      newState = state.find((meal) => meal.id === action.payload);
      newState = {
        ...newState,
        joined: true,
      };
      foundIndex = state.findIndex((x) => x.id === action.payload);
      state[foundIndex] = newState;
      return [...state];

    default:
      return state;
  }
};

// Action Creators

export const filterMeals = createAsyncThunk(FILTER_MEALS, async () => {
  const response = await fetch(API.mealEndPoint);
  const data = await response.json();

  const meals = data.map((meal) => ({
    id: meal.meal_id,
    name: meal.meal_name,
    description: meal.description,
    joined: false,
  }));

  return meals;
});

export const detailMeal = (id) => ({
  type: DETAIL_MEAL,
  payload: id,
});
