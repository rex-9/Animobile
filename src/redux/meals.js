import { createAsyncThunk } from '@reduxjs/toolkit';
import API from '../api';

const FILTER_MEALS = 'yummyvalley/meals/FILTER_MEALS';

const initialState = [];

const mealsXer = (state = initialState, action) => {
  switch (action.type) {
    case `${FILTER_MEALS}/fulfilled`:
      return [...action.payload];

    default:
      return state;
  }
};

const filterMeals = createAsyncThunk(FILTER_MEALS, async (cateName) => {
  const response = await fetch(API.filter(cateName));
  const data = await response.json();
  const meals = data.meals.map((meal) => ({
    name: meal.strMeal,
    thumb: meal.strMealThumb,
    id: meal.idMeal,
  }));
  return meals;
});

export default mealsXer;
export { filterMeals };
