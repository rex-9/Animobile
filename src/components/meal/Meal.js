import PropTypes from 'prop-types';
import { useState } from 'react';
import './Meal.css';

const Meal = ({
  index, id, name, photo,
}) => {
  const [array] = useState([]);

  for (let i = 1; i < 100; i += 1) {
    array.push(i);
    i += 4;
  }
  for (let i = 2; i < 100; i += 1) {
    array.push(i);
    i += 4;
  }

  return (
    <>
      <div className="card" style={array.includes(index) ? { backgroundColor: 'var(--dark)' } : { backgroundColor: 'var(--medium)' }}>
        <img src={photo} alt={name} />
        <div>
          <div>{name}</div>
          <div>{id}</div>
        </div>
      </div>
    </>
  );
};

Meal.propTypes = {
  index: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
};

export default Meal;
