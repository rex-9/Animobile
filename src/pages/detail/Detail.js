import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { detailMeal } from '../../redux/meal';
import Nav from '../../components/nav/Nav';
import Highlight from '../../components/highlight/Highlight';
import './Detail.css';

const Detail = () => {
  const { name, id } = useParams();
  const dispatch = useDispatch();

  const meal = useSelector((state) => state.meal);

  useEffect(() => {
    if (meal.id !== id) {
      dispatch(detailMeal(id));
    }
  }, []);

  return (
    <>
      <Nav navContent={meal.name} back route={`/${name}`} />
      <Highlight thumb={meal.photo} name={meal.name} id={meal.id} />
      <div className="detail">
        <header>
          How to prepare
        </header>
        <div>
          id:
          { meal.id }
        </div>
        <div>
          style:
          { meal.style }
        </div>
        <div>
          Category:
          { meal.category }
        </div>
        <div>
          Instructions:
          { meal.instructions }
        </div>
      </div>
    </>
  );
};

export default Detail;
