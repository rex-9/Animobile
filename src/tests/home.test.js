import {
  render, screen, act,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../pages/home/Home';
import Detail from '../pages/detail/Detail';
import { filterMeals } from '../redux/meals';
import { detailMeal } from '../redux/meal';

describe('Home page component', () => {
  afterEach(() => {
    act(() => store.dispatch(filterMeals('Pork')));
  });
  it('Check if contents of home available', async () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(screen.getByText('Seafood')).toMatchSnapshot(); // Nav component
    expect(screen.getByText('Breakfast')).toMatchSnapshot(); // Highlight component
    expect(screen.findAllByText(/WONTONS/i)).toMatchSnapshot();
  });

  // afterEach(() => {
  //   act(() => store.dispatch(detailMeal('52885')));
  // });
  it('Check if contents of Detail available', async () => {
    render(
      <Provider store={store}>
        <Detail />
      </Provider>,
    );
    expect(screen.getByText('How to prepare')).toMatchSnapshot(); // Name of the Detail component
    expect(screen.getByText('id')).toMatchSnapshot(); // id of the Detail component
    const meal = screen.findAllByText(/instructions:/i); // Instructions of the meals
    expect(meal).toMatchSnapshot();
  });
});
