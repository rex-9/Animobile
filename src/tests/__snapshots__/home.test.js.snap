// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component Check if contents of home available 1`] = `
<div>
  Seafood
</div>
`;

exports[`Home page component Check if contents of home available 2`] = `
<h1>
  Breakfast
</h1>
`;

exports[`Home page component Check if contents of home available 3`] = `Promise {}`;
